akka-actor/src/main/scala/akka/actor/ActorRef.scala

```diff
@@ -599,6 +599,23 @@ private[akka] class VirtualPathContainer(
 
   private val children = new ConcurrentHashMap[String, InternalActorRef]
 
+  override def !(message: Any)(implicit sender: ActorRef = Actor.noSender): Unit = message match {
+    case ActorSelectionMessage(msg, elements, _) ⇒ {
+      elements.foreach {
+        case SelectChildName(name) ⇒ getChild(name).tell(msg, sender)
+        case pattern @ SelectChildPattern(_) ⇒ {
+          import scala.collection.JavaConverters.mapAsScalaConcurrentMapConverter
+          children.asScala filter {
+            case (name, child) ⇒ pattern.pattern.matcher(name).matches
+          } foreach {
+            case (name, child) ⇒ child.tell(msg, sender)
+          }
+        }
+        case SelectParent ⇒ getParent.tell(msg, sender)
+      }
```

Comment:
maybe not now, but I think you should have a default case that delegates to `super.!` to avoid potential match error